tests/inc/options/option-social-test.php

```diff
@@ -0,0 +1,160 @@
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'og_default_image_id' => '' ),
+				'dirty'    => array( 'og_default_image_id' => '' ),
+				'clean'    => array( 'og_default_image_id' => '' ),
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'og_frontpage_image_id' => '' ),
+				'dirty'    => array( 'og_frontpage_image_id' => '' ),
+				'clean'    => array( 'og_frontpage_image_id' => '' ),
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'og_default_image_id' => 123 ),
+				'dirty'    => array( 'og_default_image_id' => '123' ),
+				'clean'    => array( 'og_default_image_id' => '' ),
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'og_frontpage_image_id' => 0 ),
+				'dirty'    => array( 'og_frontpage_image_id' => 'testen' ),
+				'clean'    => array( 'og_frontpage_image_id' => '' ),
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'dirty'    => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'clean'    => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'youtube_url' => 'https://www.youtube.com/yoasttube' ),
+				'dirty'    => array( 'youtube_url' => 'https://www.youtube.com/yoasttube' ),
+				'clean'    => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'old'      => array(),
+			),
+		);
+	}
+
+	/**
+	 * Data for the test_validate_option test.
+	 *
+	 * @return array The test data.
+	 */
+	public function validate_option_invalid_data_provider() {
+		return array(
+			array(
+				'expected'  => array( 'facebook_site' => '' ),
+				'dirty'     => array( 'facebook_site' => 'invalidurl' ),
+				'clean'     => array( 'facebook_site' => '' ),
+				'old'       => array(),
+				'slug_name' => 'facebook_site',
+			),
+			array(
+				'expected'  => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'dirty'     => array( 'youtube_url' => 'invalidurl' ),
+				'clean'     => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'old'       => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'slug_name' => 'youtube_url',
+			),
+		);
+	}
+
+	/**
+	 * Tests that submitting an option with an invalid URL adds a WordPress settings error notice.
+	 *
+	 * @covers WPSEO_Option_Social::validate_url
+	 */
+	public function test_validate_url_adds_settings_error() {
+		$instance = new Option_Social_Double();
+
+		Monkey\Functions\expect( 'add_settings_error' )
+			->once()
+			->with( 'yoast_wpseo_social_options', 'instagram_url', '<strong>invalidurl</strong> does not seem to be a valid url. Please correct.', 'notice-error' );
+
+		$clean = array( 'instagram_url' => '' );
+		$dirty = array( 'instagram_url' => 'invalidurl' );
+
+		$instance->validate_url( 'instagram_url', $dirty, '', $clean );
```

Comment:
I'm not sure I understand what this Brain-Monkey refactored test actually tests. 
The original version was checking the `global $wp_settings_errors` for a specific setting name and error code name, which was a way to check the presence of the specific added error in the global.

To my understanding, this new version expects `add_settings_error()` to be called once with the passed arguments. It doesn't check the actual presence of the added error in the $wp_settings_errors` array.

Aside: I'm not sure `validate_url()` should be called directly, as it's used internally by `validate_option()`.  Also, as it is now, this test seems redundant to me as it basically replicates what `test_validate_option_with_invalid_data()` already tests for (see `validate_option_invalid_data_provider`).